numba/core/serialize.py

```diff
@@ -133,6 +133,32 @@ def _rebuild_code(marshal_version, bytecode_magic, marshalled):
     return marshal.loads(marshalled)
 
 
+# Keep unpickled object via `numba_unpickle` alive.
+_unpickled_memo = {}
+
+
+def _numba_unpickle(address, bytedata, hashed):
```

Comment:
Is there merit in adding a couple of tests just to make sure memoization is working as expected?

Reply:
added a simple test in https://github.com/numba/numba/pull/5922/commits/17e72af86d927aa5d01bd7f1fbb80f8b036b0797 for the memoization of its output